contracts/upgradeable_contracts/BasicOmnibridge.sol

```diff
@@ -112,6 +111,35 @@ abstract contract BasicOmnibridge is
         _handleTokens(_token, true, _recipient, _value);
     }
 
+    /**
+     * @dev Handles the bridged tokens for the already registered token pair.
+     * Checks that the value is inside the execution limits and invokes the Mint or Unlock accordingly.
+     * @param _token address of the native ERC20/ERC677 token on the other side.
+     * @param _recipient address that will receive the tokens.
+     * @param _value amount of tokens to be received.
+     * @param _data additional transfer data passed from the other side.
+     */
+    function handleBridgedTokensAndCall(
```

Comment:
Do you think it is worth to have also an ability to call relayTokensAndCall for native tokens? For example, I would like to trade tokens on Uniswap by making a transaction in the xDai chain.

Reply:
It is indeed possible to use `relayTokensAndCall` for the xDai native token, which will result in the `transferAndCall` on the bridged token in the Mainnet.
But I think it doesn't make a lot of sense to introduce such functionality for Home Native tokens in the Foreign-to-Home direction (same for Home-to-Foreign direction for the Foreign Native tokens), since only few native tokens support ERC677.

Reply:
It is about consistency. At some point of time there will be no difference between bridged and native tokens for ordinary user: Alice can trade STAKE tokens on Uniswap in one chain and trade STAKE tokens on Honeyswap on another chain. It will be hard to explain in few monthes which tokens are native for one chain and which are not -- what is a remarkable (for ordinary user) feature to differentiate native and bridged tokens?

So, one could expect if they are able to do something through the bridge in one direction why they cannot do this in another direction, especially if tokens on both side supports the same standard (currently we know two tokens STAKE and LINK)